x-pack/legacy/plugins/canvas/public/plugin.tsx

```diff
@@ -60,6 +60,10 @@ export class CanvasPlugin
       id: 'canvas',
       title: 'Canvas App',
       async mount(context, params) {
+        // TODO: Do we want to completely move canvas_plugin_src into it's own plugin?
+        const srcPlugin = new CanvasSrcPlugin();
+        await srcPlugin.setup(core, { canvas: canvasApi });
```

Comment:
plugin `setup()` functions have a specified meaning and they shouldn't be async, if we want to follow the plugin style here I don't think we should block the canvas app mounting by downloading all the templates.